packages/orbit-design-tokens/src/dictionary/utils/__tests__/is.test.ts

```diff
@@ -58,11 +59,34 @@ describe("is utils", () => {
     expect(isModifier(tokenPlaceholder({ type: "modifier", internal: true }))).toEqual(true);
   });
   it("isInternal should return expected", () => {
-    expect(isInternal(tokenPlaceholder({ type: "color", internal: true }))).toEqual(true);
-    expect(isInternal(tokenPlaceholder({ type: "color", internal: false }))).toEqual(false);
+    expect(isInternal(tokenPlaceholder({ type: "color", internal: true }))).toBe(true);
+    expect(isInternal(tokenPlaceholder({ type: "color", internal: false }))).toBe(false);
   });
   it("isNotInternal should return expected", () => {
-    expect(isNotInternal(tokenPlaceholder({ type: "color", internal: true }))).toEqual(false);
-    expect(isNotInternal(tokenPlaceholder({ type: "color", internal: false }))).toEqual(true);
+    expect(isNotInternal(tokenPlaceholder({ type: "color", internal: true }))).toBe(false);
+    expect(isNotInternal(tokenPlaceholder({ type: "color", internal: false }))).toBe(true);
+  });
+  it("isSpacingWithTwoValues should return expected", () => {
+    expect(isSpacingWithTwoValues(["top-bottom", "left-right"])).toBe(true);
+    expect(isSpacingWithTwoValues(["left-right", "top-bottom"])).toBe(true);
+    expect(isSpacingWithTwoValues(["left-right", "bottom-top"])).toBe(false);
+    expect(isSpacingWithTwoValues(["right-left", "top-bottom"])).toBe(false);
+    expect(isSpacingWithTwoValues(["top", "right", "bottom", "left"])).toBe(false);
+    expect(isSpacingWithTwoValues(["top", "left-right", "bottom"])).toBe(false);
```

Comment:
I see your point, but I'm not sure if that's feasible currently.

One thing to consider is that these functions are being used inside the `attribute/spacing` to check if the spacing attribute is properly defined. Thus not having it would mean that the build will fail on some `map` function or similar problem.

The second thing is that even though we would aim to replace these functions with type checking, we need to still determine what kind of spacing we need to render. I took this specific approach mostly because anybody from us can define the spacing in wrongful way – either missing attribute from those three possible, or not following the order that is needed when comes to defining spacings (margin/padding) CSS attribute. Therefore these functions are also used in `flattenSpacing` function.

When comes to some examples, let's look on this:

```json
"attributes": {
  "spacing": {
    "left-right": 16,
    "top-bottom": 12
  }
}
```

If we would purely rely on type checking we will miss serializing of those attributes and thus resulting in space e.g. `16px 12px` that is incorrect. The `top` and `bottom` space would be `16px` instead of `12px` and vice-versa.

The same thing goes for this definition:

```json
"attributes": {
  "spacing": {
    "top": 16,
    "right": 12,
    "bottom": 8,
  }
}
```

In this particular example we miss the `bottom` attribute and it would be incorrect again, e.g. resulting in `16px 12px 8px`. The value of `12px` would be applied also to `left` and that can be unwanted.

Reply:
That was general description of the specificity of those functions.

When comes to usage of:

```jsx
export const isSpacingWithTwoValues = (spaceArray: ["top-bottom", "left-right"] | ["left-right", "top-bottom"]): boolean => {
  // ...
}
```

I don't think that it will be easily possible, because properties of `attributes` (and their properties) are typed currently as `type Attributes = [attribute: string]: Attributes | Array<Attributes> | string | number | boolean | undefined`.

There's no official TS defs for `style-dictionary`, but it should come with release of `3.0.0`. I have some own locally, but it's not perfect and I would like to focus on it more after migrating to this new tooling fully. Even some small refactorings will be needed as you don't see the whole picture at the start 😛